### ISSUE TITLE
Function docstring

### ISSUE THREAD

**Issue by Haxrox** [WorldofKerry/Python2Verilog#83]
**Is your feature request related to a problem? Please describe.**
It's difficult to understand what parameters are being used in functions. 

For example, in https://github.com/WorldofKerry/Python2Verilog/blob/40d923ea1de83a5505b59e690a42e6ebeb956ef6/python2verilog/optimizer/optimizer.py#L27-L30 what is `mapping`? How do I create a `mapping` to use this function? 

Additionally, what is returned by this function? Is this a valid verilog expression? a valid Python expression?

Similarly here:
https://github.com/WorldofKerry/Python2Verilog/blob/40d923ea1de83a5505b59e690a42e6ebeb956ef6/python2verilog/optimizer/optimizer.py#L52-L57

what are rvalues and what's stmt?

**Describe the solution you'd like**
A clear, concise docstring as documented here: https://google.github.io/styleguide/pyguide.html#s3.8.3-functions-and-methods

tldr a comment that describes:
1. what the function does
2. the function parameters and what they are used for
3. the functions return value and type
4. exceptions raised by the function

**Describe alternatives you've considered**
Unless these functions are used internally, you can probably get away with leaving the functions undocumented. Although, it'd be helpful for other devs to know how these functions work

**Comment by WorldofKerry** [comment]
> what is mapping? How do I create a mapping to use this function?

Not sure how I can further elaborate on that. It's a map that maps one expression to another. It can be created by using a `dict`. 

> Additionally, what is returned by this function?

I will add what a function returns to the docstrings. 

> Is this a valid verilog expression? a valid Python expression?

It is an `ir.Expression`, where the `ir` stands for Intermediate Representation, from the `ir` submodule, where the expressions are a subset of Python expressions that are synthesizable in Verilog. I will further clarify in its docstring. 

> what are rvalues and what's stmt?

That is an oversight on my part. I used `C++`-specific language thinking it was universal. I'll fix that. 

I think these functions will be used internally, so having full docstrings may not always be necessary. However, from above, I do see the importance of explaining parameters, return values, and raised exceptions, for non-trivial cases.
